dbscripts/xml/upgrade/3.2.0_preupdate_email_templates.xml

```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE data SYSTEM "../../../lib/pkp/dtd/xmlData.dtd">
+
+<!--
+  * dbscripts/xml/upgrade/3.2.0_email_templates.xml
+  *
+  * Copyright (c) 2013-2018 Simon Fraser University
+  * Copyright (c) 2003-2018 John Willinsky
+  * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+  *
+  * Update the email templates tables to remove unused columns, add a new
+	* email_id column and primary key, and update outdated role ids for the
+	* default templates.
+  *
+  -->
+
+<data>
+	<sql>
+		<query>
+			ALTER TABLE email_templates_data
+			ADD email_id BIGINT NOT NULL
+		</query>
+		<query>
+			UPDATE email_templates_data as etd
+			SET etd.email_id = (
+				SELECT et.email_id
+				FROM email_templates as et
+				WHERE etd.email_key = et.email_key
+					AND etd.assoc_type = et.assoc_type
+					AND etd.assoc_id = et.assoc_id
+			)
+		</query>
+		<query>
+			CREATE TABLE email_templates_settings (
```

Comment:
The table creation should be handled by the schema XML descriptors -- if those are listed in `dbscripts/xml/upgrade.php` then there's no need to include `CREATE TABLE` statements.

Reply:
In this case, I'm doing a little bit of extra work and I wasn't sure if it was possible to do it purely through the schema. Here's a quick rundown of what's happening:

1. Create `email_id` column in `email_templates_data`.
2. Populate that column with values from `email_templates`.
3. Use the new rows in `email_templates_data` to feed a new table, `email_templates_settings`.
4. Delete the old `email_templates_data` table.

So in this case, I can't rely on the schema to just drop and create tables, because I'm moving data over from one to the other. Is there a better way to do this?

The same is true for the `context_id` column added to `email_templates`. I add this column, then populate it with the values based on two other columns, before dropping the two other columns.